### PR TITLE
Normalize examples

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This normalizes some of the comments in the examples, as well as some of the filenames. No code changes, just wanted to make it so that the first comments in the examples weren't too descriptive (so that we wouldn't have two copies of the description to maintain) and so that they would be dartdoc comments that would link back to the thing that they are an example for.

I also renamed some files to match the naming scheme in the examples.

I had to do some surgery to `navigation_bar.dart` in the examples, since there was an example that wasn't linked, and the one that was linked was incorrectly named.

## Tests
 - There should be no change in code, just renaming and comment changes.

### CODE REVIEW

File: examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart [hunk 1]
```diff
@@ -32,14 +32,16 @@ class _DatePickerExampleState extends State<DatePickerExample> {
   DateTime time = DateTime(2016, 5, 10, 22, 35);
   DateTime dateTime = DateTime(2016, 8, 3, 17, 45);
 
-  // This shows a CupertinoModalPopup with a reasonable fixed height which hosts CupertinoDatePicker.
+  // This shows a CupertinoModalPopup with a reasonable fixed height which hosts
```
Comment (HansMuller) [inline examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart:35]: Comments can probably leave out the "This" and there's (arguably) a missing "a":
```dart
// Shows a CupertinoModalPopup with a reasonable fixed height which hosts // a CupertinoDatePicker.
```

File: examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart [hunk 2]
```diff
@@ -32,14 +32,16 @@ class _DatePickerExampleState extends State<DatePickerExample> {
...
+  // CupertinoDatePicker.
   void _showDialog(Widget child) {
     showCupertinoModalPopup<void>(
       context: context,
       builder: (BuildContext context) => Container(
         height: 216,
         padding: const EdgeInsets.only(top: 6.0),
-        // The Bottom margin is provided to align the popup above the system navigation bar.
+        // The Bottom margin is provided to align the popup above the system
```
Comment (HansMuller) [inline examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart:43]:
```suggestion
        // The bottom margin is provided to align the popup above the system
```

File: examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart [hunk 3]
```diff
@@ -85,8 +87,9 @@ class _DatePickerExampleState extends State<DatePickerExample> {
                         },
                       ),
                     ),
-                    // In this example, the date value is formatted manually. You can use intl package
-                    // to format the value based on user's locale settings.
+                    // In this example, the date value is formatted manually.
```
Comment (HansMuller) [inline examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart:90]:
```suggestion
                    // In this example, the date is formatted manually.
```

File: examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart [hunk 4]
```diff
@@ -85,8 +87,9 @@ class _DatePickerExampleState extends State<DatePickerExample> {
...
+                    // You can use intl package to format the value based on
```
Comment (HansMuller) [inline examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart:91]:
```suggestion
                    // You can use the intl package to format the value based on the
```

File: examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart [hunk 5]
```diff
@@ -111,8 +114,9 @@ class _DatePickerExampleState extends State<DatePickerExample> {
                         },
                       ),
                     ),
-                    // In this example, the time value is formatted manually. You can use intl package to
-                    // format the value based on the user's locale settings.
+                    // In this example, the time value is formatted manually.
+                    // You can use intl package to format the value based on the
```
Comment (HansMuller) [inline examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart:118]:
```suggestion
                    // You can use the intl package to format the value based on the
```

File: examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart [hunk 6]
```diff
@@ -136,8 +140,9 @@ class _DatePickerExampleState extends State<DatePickerExample> {
                         },
                       ),
                     ),
-                    // In this example, time value is formatted manually. You can use intl package to
-                    // format the value based on the user's locale settings.
+                    // In this example, time value is formatted manually. You
```
Comment (HansMuller) [inline examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart:143]:
```suggestion
                    // In this example, the time value is formatted manually. You
```

File: examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart [hunk 7]
```diff
@@ -136,8 +140,9 @@ class _DatePickerExampleState extends State<DatePickerExample> {
...
+                    // can use intl package to format the value based on the
```
Comment (HansMuller) [inline examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart:144]:
```suggestion
                    // can use the intl package to format the value based on the
```

File: examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart [hunk 8]
```diff
@@ -30,14 +30,16 @@ class TimerPickerExample extends StatefulWidget {
 class _TimerPickerExampleState extends State<TimerPickerExample> {
   Duration duration = const Duration(hours: 1, minutes: 23);
 
-  // This shows a CupertinoModalPopup with a reasonable fixed height which hosts CupertinoTimerPicker.
+  // This shows a CupertinoModalPopup with a reasonable fixed height which hosts
```
Comment (HansMuller) [inline examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart:33]:
```suggestion
  // Shows a CupertinoModalPopup with a reasonable fixed height which hosts
```

File: examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart [hunk 9]
```diff
@@ -30,14 +30,16 @@ class TimerPickerExample extends StatefulWidget {
...
+  // CupertinoTimerPicker.
```
Comment (HansMuller) [inline examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart:34]:
```suggestion
  // a CupertinoTimerPicker.
```

File: examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart [hunk 10]
```diff
@@ -30,14 +30,16 @@ class TimerPickerExample extends StatefulWidget {
...
   void _showDialog(Widget child) {
     showCupertinoModalPopup<void>(
       context: context,
       builder: (BuildContext context) => Container(
         height: 216,
         padding: const EdgeInsets.only(top: 6.0),
-        // The Bottom margin is provided to align the popup above the system navigation bar.
+        // The Bottom margin is provided to align the popup above the system
```
Comment (HansMuller) [inline examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart:41]:
```suggestion
        // The bottom margin is provided to align the popup above the system
```

File: examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart [hunk 11]
```diff
@@ -76,14 +78,16 @@ class _TimerPickerExampleState extends State<TimerPickerExample> {
                       CupertinoTimerPicker(
                         mode: CupertinoTimerPickerMode.hm,
                         initialTimerDuration: duration,
-                        // This is called when the user changes the timer duration.
+                        // This is called when the user changes the timer
```
Comment (HansMuller) [inline examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart:81]:
```suggestion
                        // This is called when the user changes the timer's
```

File: examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart [hunk 12]
```diff
@@ -76,14 +78,16 @@ class _TimerPickerExampleState extends State<TimerPickerExample> {
...
+                        // duration.
                         onTimerDurationChanged: (Duration newDuration) {
                           setState(() => duration = newDuration);
                         },
                       ),
                     ),
-                    // In this example, the timer value is formatted manually. You can use intl package
-                    // to format the value based on user's locale settings.
+                    // In this example, the timer value is formatted manually.
```
Comment (HansMuller) [inline examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart:88]:
```suggestion
                    // In this example, the timer's value is formatted manually.
```

File: examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart [hunk 13]
```diff
@@ -76,14 +78,16 @@ class _TimerPickerExampleState extends State<TimerPickerExample> {
...
+                    // You can use intl package to format the value based on
```
Comment (HansMuller) [inline examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart:89]:
```suggestion
                    // You can use the intl package to format the value based on
```

File: examples/api/lib/foundation/change_notifier/change_notifier.0.dart [hunk 14]
```diff
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-// Flutter code sample for ChangeNotifier with an AnimatedBuilder
+/// Flutter code sample for [ChangeNotifier] with an AnimatedBuilder.
```
Comment (HansMuller) [inline examples/api/lib/foundation/change_notifier/change_notifier.0.dart:5]:
```suggestion
/// Flutter code sample for a [ChangeNotifier] with an AnimatedBuilder.
```

File: examples/api/lib/material/action_chip/action_chip.0.dart [hunk 15]
```diff
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-// Flutter code sample ActionChip
+// Flutter code sample ActionChip.
```
Comment (HansMuller) [inline examples/api/lib/material/action_chip/action_chip.0.dart:5]:
```suggestion
// Flutter code sample for [ActionChip].
```

File: examples/api/lib/material/filter_chip/filter_chip.0.dart [hunk 16]
```diff
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-// Flutter code sample FilterChip
+// Flutter code sample FilterChip.
```
Comment (HansMuller) [inline examples/api/lib/material/filter_chip/filter_chip.0.dart:5]:
```suggestion
// Flutter code sample for [FilterChip].
```

File: examples/api/lib/widgets/animated_list/animated_list.0.dart [hunk 17]
```diff
@@ -50,11 +50,10 @@ class _AnimatedListSampleState extends State<AnimatedListSample> {
...
   }
 
-  // Used to build an item after it has been removed from the list. This
-  // method is needed because a removed item remains visible until its
-  // animation has completed (even though it's gone as far this ListModel is
-  // concerned). The widget will be used by the
-  // [AnimatedListState.removeItem] method's
+  // Used to build an item after it has been removed from the list. This method
+  // is needed because a removed item remains visible until its animation has
+  // completed (even though it's gone as far this ListModel is concerned). The
```
Comment (HansMuller) [inline examples/api/lib/widgets/animated_list/animated_list.0.dart:55]:
```suggestion
  // completed (even though it's gone as far as this ListModel is concerned). The
```

File: examples/api/lib/widgets/hardware_keyboard/key_event_manager.0.dart [hunk 18]
```diff
@@ -53,7 +51,7 @@ class FallbackDemoState extends State<FallbackDemo> {
         constraints: const BoxConstraints(maxWidth: 500, maxHeight: 400),
         child: Column(
           children: <Widget>[
-            const Text('This area handles key pressses that are unhandled by any shortcuts by displaying them below. '
+            const Text('This area handles key presses that are unhandled by any shortcuts by displaying them below. '
```
Comment (HansMuller) [inline examples/api/lib/widgets/hardware_keyboard/key_event_manager.0.dart:54]:
```suggestion
            const Text('This area handles key presses that are unhandled by any shortcuts, by displaying them below. '
```

File: examples/api/lib/widgets/routes/popup_route.0.dart [hunk 19]
```diff
@@ -42,8 +42,8 @@ class DismissibleDialog<T> extends PopupRoute<T> {
   @override
   Color? get barrierColor => Colors.black.withAlpha(0x50);
 
-  /// This allows the popup to be dismissed by tapping the scrim or by
-  /// pressing escape key on the keyboard.
+  // This allows the popup to be dismissed by tapping the scrim or by pressing
+  // escape key on the keyboard.
```
Comment (HansMuller) [inline examples/api/lib/widgets/routes/popup_route.0.dart:46]:
```suggestion
  // the escape key on the keyboard.
```

File: examples/api/lib/widgets/shared_app_data/shared_app_data.0.dart [hunk 20]
```diff
@@ -13,16 +13,16 @@ class ShowSharedValue extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    // The SharedAppData.getValue() call here causes this widget to depend
-    // on the value of the SharedAppData's 'foo' key. If it's changed, with
+    // The SharedAppData.getValue() call here causes this widget to depend on
```
Comment (HansMuller) [inline examples/api/lib/widgets/shared_app_data/shared_app_data.0.dart:16]:
```suggestion
    // The SharedAppData.getValue() call causes this widget to depend on
```

**Review by HansMuller — APPROVED** [review]
LGTM

File: examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart (hunk 1 above)
Comment (gspencergoog) [reply examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart:35]: Well, but without the "This", it's a sentence fragment, not a sentence, and we have a style guide rule that says to use correct grammar, with almost exactly this scenario as an example. I mean, sure, it's talking about capitalization, but it's still incorrect grammar to use sentence fragments.

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use-correct-grammar

File: examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart (hunk 6 above)
Comment (gspencergoog) [reply examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart:143]: Done

File: examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart (hunk 7 above)
Comment (gspencergoog) [reply examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart:144]: Done

File: examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart (hunk 11 above)
Comment (gspencergoog) [reply examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart:81]: Done.

File: examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart (hunk 12 above)
Comment (gspencergoog) [reply examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart:88]: Done.

File: examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart (hunk 13 above)
Comment (gspencergoog) [reply examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart:89]: Done.

File: examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart (hunk 10 above)
Comment (gspencergoog) [reply examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart:41]: Done.

File: examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart (hunk 9 above)
Comment (gspencergoog) [reply examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart:34]: Done.

File: examples/api/lib/foundation/change_notifier/change_notifier.0.dart (hunk 14 above)
Comment (gspencergoog) [reply examples/api/lib/foundation/change_notifier/change_notifier.0.dart:5]: Done.

File: examples/api/lib/material/action_chip/action_chip.0.dart (hunk 15 above)
Comment (gspencergoog) [reply examples/api/lib/material/action_chip/action_chip.0.dart:5]: Nice!  I missed that one!

File: examples/api/lib/material/filter_chip/filter_chip.0.dart (hunk 16 above)
Comment (gspencergoog) [reply examples/api/lib/material/filter_chip/filter_chip.0.dart:5]: Done.

File: examples/api/lib/widgets/animated_list/animated_list.0.dart (hunk 17 above)
Comment (gspencergoog) [reply examples/api/lib/widgets/animated_list/animated_list.0.dart:55]: Done.

File: examples/api/lib/widgets/hardware_keyboard/key_event_manager.0.dart (hunk 18 above)
Comment (gspencergoog) [reply examples/api/lib/widgets/hardware_keyboard/key_event_manager.0.dart]: Done.

File: examples/api/lib/widgets/routes/popup_route.0.dart (hunk 19 above)
Comment (gspencergoog) [reply examples/api/lib/widgets/routes/popup_route.0.dart]: Done.

File: examples/api/lib/widgets/shared_app_data/shared_app_data.0.dart (hunk 20 above)
Comment (gspencergoog) [reply examples/api/lib/widgets/shared_app_data/shared_app_data.0.dart]: Done.

File: examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart (hunk 1 above)
Comment (gspencergoog) [reply examples/api/lib/cupertino/date_picker/cupertino_date_picker.0.dart:35]: It's just as grammatically correct, but it's probably a better sentence. I'll add that.

**Pull request merged**